orca-sql/src/main/kotlin/com/netflix/spinnaker/orca/sql/pipeline/persistence/SqlExecutionRepository.kt

```diff
@@ -783,7 +783,7 @@ class SqlExecutionRepository(
   private fun storeExecutionInternal(ctx: DSLContext, execution: PipelineExecution, storeStages: Boolean = false) {
     validateHandledPartitionOrThrow(execution)
 
-    val pipelineTrigger = mutatePipelineTrigger(execution)
+    execution.trigger = mapper.convertValue(execution.trigger, Trigger::class.java)
```

Comment:
:+1: on utilizing converters. This is the perfect use case for it. However, Id register it on the mapper instead of calling it directly. That way this code doesn't change.

What needs to happen here is a bean needs to be specific for the sql execution repository. This bean must register **two** converters.

1st converter will convert `PipelineTrigger` -> `PipelineTriggerRef`
2nd converter goes the other way `PipelineTriggerRef` -> `PipelineTrigger`

If we go with this approach we may not even need to touch that crazy trigger deserializer and do things much betteer.